protocols/ledger/ledger.go

```diff
@@ -0,0 +1,141 @@
+package ledger
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/statechannels/go-nitro/channel"
+	"github.com/statechannels/go-nitro/channel/state"
+	"github.com/statechannels/go-nitro/channel/state/outcome"
+	"github.com/statechannels/go-nitro/protocols"
+	"github.com/statechannels/go-nitro/types"
+)
+
+type LedgerManager struct {
+	nonce *big.Int
+}
+
+func NewLedgerManager() LedgerManager {
```

Comment:
The Ledger Manager could accept a `store` and read and write from the `store` itself but that would break with previous patterns where the `engine` is soley responsible for fetching and writing to the store.